src/main/java/seedu/address/model/human/person/Person.java

```diff
@@ -41,6 +41,27 @@ public Person(Name name, Phone phone, Address address, TripDay tripDay, TripTime
         this.tags.addAll(tags);
     }
 
+    /**
+     * Creates a new {@code Person} with a driver.
+     * @param name the {@code Name} of the {@code Person}
+     * @param phone the {@code Phone} of the {@code Person}
+     * @param address the {@code Address} of the {@code Person}
+     * @param tripDay the {@code TripDay} of the {@code Person}
+     * @param tripTime the {@code TripTime} of the {@code Person}
+     * @param driver the {@code Driver} assigned to {@code Person}
+     * @param tags the {@code Tag}s of the {@code Person}
+     */
+    public Person(Name name, Phone phone, Address address, TripDay tripDay, TripTime tripTime, Driver driver,
+                  Set<Tag> tags) {
+        super(name, phone);
+        requireAllNonNull(address, tripDay, tripTime, tags);
+        this.address = address;
+        this.tripDay = tripDay;
+        this.tripTime = tripTime;
+        this.driver = Optional.of(driver);
+        this.tags.addAll(tags);
+    }
+
```

Comment:
should we have an overloaded constructor?

Reply:
It might be needed as we should allow the user to add themselves to the system regardless of whether they have been assigned a driver or not